### ISSUE TITLE
WordPress APO not compatible with LiteSpeed Cache for WordPress

### ISSUE THREAD

**Issue by colinmcdermott** [cloudflare/Cloudflare-WordPress#346]
Hi guys,

Are there any plans to make the WP APO compatible Litespeed Cache, or any other caching plugins?

Currently it seems that CF ignores most LSCWP optimisations, and duplicates all the JS/CSS files.

So for example if you use LSCWP to minify and combine CSS into a single file, CF APO will ignore this, and replicate all the CSS files in full.

So my site goes from having 4-5 resources, to 50+ on each page.

I assume it will be the same case with all caching plugins but I have not checked this.

Similarly I use the very popular AutoOptimize plugin to ensure we are not loading CSS/JS files on pages that don't need them. I expect this functionality is broken also.

Is this something you guys plan to work on? Or would it be fair to say that this service is not suitable for sites that conduct their own web perf optimisations?

**Comment by sejoker** [comment]
Hi,

I posted answer to the similar question here: https://community.cloudflare.com/t/apo-with-another-caching-plugin-that-purge-cache/233566/5?u=yevgen

TLDR version:
purge cache via Cloudflare Purge API for the selected page one more time once LSCWP or AutoOptimize finished updating the  page.

**Comment by colinmcdermott** [comment]
Hi, thanks for the reply @sejoker.

Have as anyone confirmed this works? Because after dozens of purged caches in every combination I can think of, all resources are still currently used by CloudFlare:

https://gtmetrix.com/reports/www.searchcandy.uk/ABSxex57/ / https://i.imgur.com/qdtDcQd.png

This is a sitewide issue that affects every page of my site on a permanent basis.

Are there any caching plugins that have been confirmed to work with the APO?

**Comment by sejoker** [comment]
The origin response and cached version of your site are identical, you can test it yourself with hard refresh in browser or passing "Cache-control: no-cache" request header.

**Comment by colinmcdermott** [comment]
From testing it seems that the issue is at least partially being caused by the CloudFlare plugin itself.

On switching CF to DNS only mode, the issue still remained. The CF plugin was still installed, so deactivated that and the issue is fully resolved.

I believe I need to have the CF plugin installed to use the APO?

**Comment by colinmcdermott** [comment]
Litespeed Cache for WP does have a CF API connection section. Is it possible to set up APO just using this, minus the CF plugin?

**Comment by colinmcdermott** [comment]
Assume this is not something that is going to be fixed anytime soon so am cancelling the APO on my account. But I would like to get this resolved for others and come back to test again at a later date.

**Comment by colinmcdermott** [comment]
Just thought I would make you guys aware that CloudFlare support are closing tickets for queries like this. I would have thought you guys would be interested to engage with people helping you test out your new product... but I guess not.

    Hi there,
    
    My name is ****** with Cloudflare Support, thanks for following up with us on this issue. We do not consider an incompatibility issue to be a technical issue with Cloudflare, as we have a list of plugins which are compatible, and consider this a user misconfiguration error.
    
    I will mark this as closed, thank you very much for contacting Cloudflare Support.
    
    Take care,

**Comment by sejoker** [comment]
APO is compatible with LiteSpeed Cache when only LSP caching is used, all the remaining LSP content optimizations require triggering CF cache purge manually as explained here:  https://blog.cloudflare.com/apo-post-launch-report/#caching-and-optimizations-plugins
